package.json

```diff
@@ -59,7 +64,10 @@
     "_test": "jest",
     "test": "turbo _test",
     "typecheck": "turbo typecheck",
-    "validate": "turbo format.cargo format.forge format.prettier lint.cargo lint.eslint lint.solhint build _compile --cache-dir=.turbo"
+    "validate": "turbo format.cargo format.forge format.prettier lint.cargo lint.eslint lint.solhint build _compile --cache-dir=.turbo",
+    "githook:precommit": "lint-staged && pnpm run typecheck",
+    "githook:prepush": "pnpm run _test --coverage",
```

Comment:
I don't like the `prepush` hook so much.
tests can be a bit long / slow to run.
I'd prefer delegating running the test to the github runners.
Moreover, as I understand it, this prepush hook will run on changes on ANY files. Which is not ideal (think e.g about .md files...)
With github workflows you can specify which files should trigger the workflow.

Reply:
Well the idea behind it is to catch the error locally before we push and wait the github action to fail and then try to resolve to finally push right code again. Its a health check that we should do locally for our changes because maybe they could be affecting other code unintentionally. Also our tests are not too many, and they are talking locally some seconds to finish. 
And I agree with you about the point that this prepush triggers on any commit changes, I haven't searched on this yet but I think we can ignore that somehow.

Reply:
I see your point, I still think that it is useful to check my tests locally before pushing but as elaborated its a different workflow that it is not the case with AnonKlub and also we are a small team. 
Since we didn't come up with an agreement on this I will revert the `pre-push` hook and only leave the `pre-commit` and the `commit-msg`